<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dlg_Manu</class>
 <widget class="QDialog" name="Dlg_Manu">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>637</width>
    <height>426</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <property name="styleSheet">
   <string notr="true">*{
font: 14pt;
font-family: &quot;微软雅黑&quot;;
}

QWidget {
color: #c2c2c3;
background: #343639;
}

QDialog {
border:1px solid black;
border-radius:1px;
}

#titleFrame {
border: no;
min-height: 20px;
background: url(&quot;:/Res/image/title_bar.png&quot;);
}

QTabWidget {
outline: no;
}

QTabWidget::tab-bar {
	left: 15px;
}

QTabWidget::pane {
border: 1px solid black;
}

QTabBar::tab{
background: #343639;
border: 1px solid black;
border-bottom-color: #343639; 
border-top-left-radius: 4px;
border-top-right-radius: 4px;
min-width: 70px;
padding: 2px;
min-height: 15px;
}

QTabBar::tab:selected{
color: white;
outline: no;
padding-top: 4px;
background: #2773b4;
border-bottom: #2773b4;
}

QTabBar::tab:!selected {
 margin-top: 2px; /* make non-selected tabs look smaller */

}

#ServoFrame {
border: no;
}

QPushButton {
border: no;
font: 16pt;
outline: no;
min-width: 80px;
min-height: 50px;
image: url(&quot;:/Res/image/button_80x50_normal.png&quot;);
}

QPushButton:pressed {
color: black;
font: 16pt;
border: no;
min-width: 80px;
min-height: 50px;
image: url(&quot;:/Res/image/button_80x50_active.png&quot;);
}
/*
QPushButton {
border: no;
outline: no;
min-width: 50px;
min-height: 50px;
image: url(&quot;:/Res/image/button_50x50_normal.png&quot;);
}

QPushButton:pressed {
color: black;
border: no;
min-width: 50px;
min-height: 50px;
image: url(&quot;:/Res/image/button_50x50_active.png&quot;);
}
*/
QPushButton:checked {
color: black;
border: no;
min-width: 80px;
min-height: 50px;
image: url(&quot;:/Res/image/button_80x50_active.png&quot;);
}

QPushButton[objectName=&quot;btnRun&quot;] {
border: no;
min-width: 80px;
min-height: 50px;
image: url(&quot;:/Res/image/btn_manual_run_80x50_normal.png&quot;);
}

QPushButton[objectName=&quot;btnRun&quot;]:pressed {
color: black;
border: no;
min-width: 80px;
min-height: 50px;
image: url(&quot;:/Res/image/btn_manual_run_80x50_active.png&quot;);
}

QPushButton[objectName=&quot;btnRun&quot;]:checked {
color: black;
border: no;
min-width: 80px;
min-height: 50px;
image: url(&quot;:/Res/image/btn_manual_run_80x50_active.png&quot;);
}

QPushButton[objectName=&quot;btnStp&quot;] {
border: no;
min-width: 80px;
min-height: 50px;
image: url(&quot;:/Res/image/btn_manual_stop_80x50_normal.png&quot;);
}

QPushButton[objectName=&quot;btnStp&quot;]:pressed {
color: black;
border: no;
min-width: 80px;
min-height: 50px;
image: url(&quot;:/Res/image/btn_manual_stop_80x50_active.png&quot;);
}

QPushButton[objectName=&quot;btnOk&quot;] {
border: no;
min-width: 80px;
min-height: 50px;
image: url(&quot;:/Res/image/button_80x50_normal.png&quot;);
}

QPushButton[objectName=&quot;btnOk&quot;]:pressed {
color: black;
border: no;
min-width: 80px;
min-height: 50px;
image: url(&quot;:/Res/image/button_80x50_active.png&quot;);
}

QLineEdit {
color: black;
background: qlineargradient(
  x1:0, y1:0, x2:0, y2:1,
  stop:0 gray,
  stop: 0.2 white
  stop:1 white
 );
border-radius: 1px;
border: 1px solid black;
min-height: 30px;
width: 150px
}

QGroupBox {
border: 1px solid black;
border-radius: 1px
}

QCheckBox {
outline: no;
     spacing: 5px;
}

QCheckBox::indicator {
     width: 50px;
     height: 40px;
}

QCheckBox::indicator:unchecked {
     image: url(&quot;:/Res/image/btn_cb_slider_45x30_off.png&quot;);
}

QCheckBox::indicator:checked {
     image: url(&quot;:/Res/image/btn_cb_slider_45x30_on.png&quot;);
}

QRadioButton {
outline: no;
spacing: 5px;
}

QRadioButton::indicator {
     width: 28px;
     height: 28px;
}

QRadioButton::indicator:unchecked {
     image: url(&quot;:/Res/image/btn_rb_28x28_off.png&quot;);
}

QRadioButton::indicator:checked {
     image: url(&quot;:/Res/image/btn_rb_28x28_on.png&quot;);
}</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_8">
   <property name="spacing">
    <number>6</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QFrame" name="titleFrame">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_7">
     <property name="leftMargin">
      <number>9</number>
     </property>
     <property name="rightMargin">
      <number>9</number>
     </property>
     <property name="bottomMargin">
      <number>9</number>
     </property>
     <item>
      <widget class="QTabWidget" name="tabWidget">
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="tab_servo">
        <attribute name="title">
         <string>伺服</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_4" stretch="0,0,1">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout">
             <property name="spacing">
              <number>50</number>
             </property>
             <property name="leftMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QPushButton" name="btnServoX">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="minimumSize">
                <size>
                 <width>80</width>
                 <height>50</height>
                </size>
               </property>
               <property name="text">
                <string>X1</string>
               </property>
               <property name="checkable">
                <bool>true</bool>
               </property>
               <property name="checked">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btnServoY1">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>Z1</string>
               </property>
               <property name="checkable">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btnServoZ1">
               <property name="enabled">
                <bool>true</bool>
               </property>
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>U</string>
               </property>
               <property name="checkable">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btnServoY2">
               <property name="enabled">
                <bool>true</bool>
               </property>
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>X2</string>
               </property>
               <property name="checkable">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btnServoZ2">
               <property name="enabled">
                <bool>true</bool>
               </property>
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>Z2</string>
               </property>
               <property name="checkable">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="Line" name="line_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <item>
              <layout class="QVBoxLayout" name="verticalLayout_2">
               <property name="leftMargin">
                <number>0</number>
               </property>
               <item>
                <widget class="QPushButton" name="btnSpeedHigh">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>高速</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="btnSpeedLow">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>低速</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QProgressBar" name="pbMSpd">
               <property name="value">
                <number>24</number>
               </property>
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeType">
                <enum>QSizePolicy::Expanding</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>50</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <layout class="QVBoxLayout" name="verticalLayout">
               <property name="spacing">
                <number>20</number>
               </property>
               <property name="leftMargin">
                <number>0</number>
               </property>
               <property name="topMargin">
                <number>10</number>
               </property>
               <property name="bottomMargin">
                <number>10</number>
               </property>
               <item>
                <layout class="QHBoxLayout" name="horizontalLayout_5">
                 <item>
                  <widget class="QRadioButton" name="rbAbsPos">
                   <property name="sizePolicy">
                    <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
                     <horstretch>0</horstretch>
                     <verstretch>0</verstretch>
                    </sizepolicy>
                   </property>
                   <property name="text">
                    <string>绝对位置</string>
                   </property>
                   <property name="checked">
                    <bool>true</bool>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QRadioButton" name="rbRelPos">
                   <property name="sizePolicy">
                    <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
                     <horstretch>0</horstretch>
                     <verstretch>0</verstretch>
                    </sizepolicy>
                   </property>
                   <property name="text">
                    <string>相对位置</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
               <item>
                <widget class="TLineEdit" name="lePos">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="layoutDirection">
                  <enum>Qt::LeftToRight</enum>
                 </property>
                </widget>
               </item>
               <item>
                <layout class="QHBoxLayout" name="horizontalLayout_2">
                 <property name="spacing">
                  <number>24</number>
                 </property>
                 <property name="leftMargin">
                  <number>0</number>
                 </property>
                 <property name="rightMargin">
                  <number>0</number>
                 </property>
                 <item>
                  <widget class="QPushButton" name="btnRun">
                   <property name="sizePolicy">
                    <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
                     <horstretch>0</horstretch>
                     <verstretch>0</verstretch>
                    </sizepolicy>
                   </property>
                   <property name="text">
                    <string/>
                   </property>
                   <property name="checkable">
                    <bool>false</bool>
                   </property>
                   <property name="autoDefault">
                    <bool>true</bool>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QPushButton" name="btnStp">
                   <property name="sizePolicy">
                    <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
                     <horstretch>0</horstretch>
                     <verstretch>0</verstretch>
                    </sizepolicy>
                   </property>
                   <property name="text">
                    <string/>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </item>
             <item>
              <spacer name="horizontalSpacer_2">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <layout class="QVBoxLayout" name="verticalLayout_6">
               <item>
                <widget class="QPushButton" name="btnUp">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>+</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="btnDwn">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>-</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="tab_IO">
        <attribute name="title">
         <string>端口</string>
        </attribute>
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0">
          <widget class="QCheckBox" name="cbO1">
           <property name="text">
            <string>输出1</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QCheckBox" name="cbO7">
           <property name="text">
            <string>输出7</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QCheckBox" name="cbO13">
           <property name="text">
            <string>输出13</string>
           </property>
          </widget>
         </item>
         <item row="0" column="3">
          <widget class="QCheckBox" name="cbO19">
           <property name="text">
            <string>输出19</string>
           </property>
          </widget>
         </item>
         <item row="0" column="4">
          <widget class="QCheckBox" name="cbO25">
           <property name="text">
            <string>输出25</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QCheckBox" name="cbO2">
           <property name="text">
            <string>输出2</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QCheckBox" name="cbO8">
           <property name="text">
            <string>输出8</string>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QCheckBox" name="cbO14">
           <property name="text">
            <string>输出14</string>
           </property>
          </widget>
         </item>
         <item row="1" column="3">
          <widget class="QCheckBox" name="cbO20">
           <property name="text">
            <string>输出20</string>
           </property>
          </widget>
         </item>
         <item row="1" column="4">
          <widget class="QCheckBox" name="cbO26">
           <property name="text">
            <string>输出26</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QCheckBox" name="cbO3">
           <property name="text">
            <string>输出3</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QCheckBox" name="cbO9">
           <property name="text">
            <string>输出9</string>
           </property>
          </widget>
         </item>
         <item row="2" column="2">
          <widget class="QCheckBox" name="cbO15">
           <property name="text">
            <string>输出15</string>
           </property>
          </widget>
         </item>
         <item row="2" column="3">
          <widget class="QCheckBox" name="cbO21">
           <property name="text">
            <string>输出21</string>
           </property>
          </widget>
         </item>
         <item row="2" column="4">
          <widget class="QCheckBox" name="cbO27">
           <property name="text">
            <string>Z1刹车</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QCheckBox" name="cbO4">
           <property name="text">
            <string>输出4</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QCheckBox" name="cbO10">
           <property name="text">
            <string>输出10</string>
           </property>
          </widget>
         </item>
         <item row="3" column="2">
          <widget class="QCheckBox" name="cbO16">
           <property name="text">
            <string>输出16</string>
           </property>
          </widget>
         </item>
         <item row="3" column="3">
          <widget class="QCheckBox" name="cbO22">
           <property name="text">
            <string>输出22</string>
           </property>
          </widget>
         </item>
         <item row="3" column="4">
          <widget class="QCheckBox" name="cbO28">
           <property name="text">
            <string>Z2刹车</string>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QCheckBox" name="cbO5">
           <property name="text">
            <string>输出5</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QCheckBox" name="cbO11">
           <property name="text">
            <string>输出11</string>
           </property>
          </widget>
         </item>
         <item row="4" column="2">
          <widget class="QCheckBox" name="cbO17">
           <property name="text">
            <string>输出17</string>
           </property>
          </widget>
         </item>
         <item row="4" column="3">
          <widget class="QCheckBox" name="cbO23">
           <property name="text">
            <string>输出23</string>
           </property>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QCheckBox" name="cbO6">
           <property name="text">
            <string>输出6</string>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <widget class="QCheckBox" name="cbO12">
           <property name="text">
            <string>输出12</string>
           </property>
          </widget>
         </item>
         <item row="5" column="2">
          <widget class="QCheckBox" name="cbO18">
           <property name="text">
            <string>输出18</string>
           </property>
          </widget>
         </item>
         <item row="5" column="3">
          <widget class="QCheckBox" name="cbO24">
           <property name="text">
            <string>输出24</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>6</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QPushButton" name="btnReset">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>复位</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_3">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="btnOk">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>确定</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnCancel">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>取消</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>TLineEdit</class>
   <extends>QLineEdit</extends>
   <header>tlineedit.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
